compiler/noirc_evaluator/src/ssa/ssa_gen/program.rs

```diff
@@ -11,13 +12,16 @@ use crate::ssa::ir::{
 };
 use noirc_frontend::hir_def::types::Type as HirType;
 
+use super::ValueId;
+
 /// Contains the entire SSA representation of the program.
 #[serde_as]
 #[derive(Serialize, Deserialize)]
 pub(crate) struct Ssa {
     #[serde_as(as = "Vec<(_, _)>")]
     pub(crate) functions: BTreeMap<FunctionId, Function>,
     pub(crate) globals: Function,
+    pub(crate) used_global_values: HashSet<ValueId>,
```

Comment:
Could we not mutate the globals Function to remove all unused values?

Reply:
It's not clear to me why we need to continue to track these unused globals

Reply:
Yeah I wanted to do the same. But globals iterate over values (as we want to convert constants to Brillig opcodes not just instructions). Removing these would require a new map structure for globals: https://github.com/noir-lang/noir/blob/ad5a9808ab7c2a89beadb6f95c664441ee27a5f8/compiler/noirc_evaluator/src/ssa/ir/map.rs#L171